File: game/src/main/java/com/bship/games/endpoints/board/errors/validations/bulk/PlacementExistenceCheckValidation.java
package com.bship.games.endpoints.board.errors.validations.bulk;

import com.bship.games.endpoints.cabinet.entity.Piece;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.util.List;
import java.util.Objects;

public class PlacementExistenceCheckValidation implements ConstraintValidator<PlacementExistenceCheck, List<Piece>> {
   public void initialize(PlacementExistenceCheck constraint) {
   }

   public boolean isValid(List<Piece> pieces, ConstraintValidatorContext context) {
      return pieces.stream().map(Piece::getPlacement).allMatch(Objects::nonNull);
   }
}

File: game/src/main/java/com/bship/games/endpoints/board/errors/validations/bulk/PlacementExistenceCheckValidation.kt
package com.bship.games.endpoints.board.errors.validations.bulk

import com.bship.games.endpoints.cabinet.entity.Piece
import java.util.*
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class PlacementExistenceCheckValidation : ConstraintValidator<PlacementExistenceCheck, List<Piece>> {
    override fun initialize(constraint: PlacementExistenceCheck) {}

    override fun isValid(pieces: List<Piece>, context: ConstraintValidatorContext): Boolean {
        return pieces.map({ it.placement }).none { obj -> Objects.isNull(obj) || Objects.isNull(obj.x) && Objects.isNull(obj.y) }

    }
}
